src/blueprint/azext_blueprint/custom.py

```diff
@@ -0,0 +1,580 @@
+                         published_blueprint=body)
+
+
+def delete_blueprint_version(cmd, client, blueprint_name, version_id, management_group=None, scope=None):
+    return client.delete(scope=scope,
+                         blueprint_name=blueprint_name,
+                         version_id=version_id)
+
+
+def get_blueprint_version(cmd, client, blueprint_name, version_id, management_group=None, scope=None):
+    return client.get(scope=scope,
+                      blueprint_name=blueprint_name,
+                      version_id=version_id)
+
+
+def list_blueprint_version(cmd, client, blueprint_name, management_group=None, scope=None):
+    return client.list(scope=scope, blueprint_name=blueprint_name)
+
+
+def get_blueprint_version_artifact(cmd, client, blueprint_name,
+                                   version_id, artifact_name, management_group=None, scope=None):
+    return client.get(scope=scope,
+                      blueprint_name=blueprint_name,
+                      version_id=version_id,
+                      artifact_name=artifact_name)
+
+
+def list_blueprint_version_artifact(cmd, client, blueprint_name,
+                                    version_id, management_group=None, scope=None):
+    return client.list(scope=scope,
+                       blueprint_name=blueprint_name,
+                       version_id=version_id)
+
+
+def _del_artifact_name(rg):
+    del rg['artifact_name']
+    return rg
+
+
+def create_blueprint_assignment(cmd,
+                                client,
+                                assignment_name,
+                                identity_type,
+                                management_group=None,
+                                scope=None,
+                                location=None,
+                                resource_groups=None,
+                                identity_principal_id=None,
+                                identity_tenant_id=None,
+                                identity_user_assigned_identities=None,
+                                display_name=None,
+                                description=None,
+                                blueprint_id=None,
+                                locks_mode=None,
+                                locks_excluded_principals=None,
+                                parameters=None):
+    body = {}
+    body['location'] = location  # str
+    body.setdefault('identity', {})['type'] = identity_type  # str
+    body.setdefault('identity',
+                    {})['principal_id'] = identity_principal_id  # str
+    body.setdefault('identity', {})['tenant_id'] = identity_tenant_id  # str
+    body.setdefault(
+        'identity', {}
+    )['user_assigned_identities'] = identity_user_assigned_identities  # dictionary
+    body['display_name'] = display_name  # str
+    body['description'] = description  # str
+    body['blueprint_id'] = blueprint_id  # str
+    body['parameters'] = json.loads(

```

Comment:
same as above, json load exception catch/rethrow

Reply:
Moved the json.loads() logic to validator.